src/test/java/com/microsoft/spring/data/gremlin/common/GremlinFactoryUnitTest.java

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.spring.data.gremlin.common;
+
+import com.microsoft.spring.data.gremlin.exception.IllegalGremlinConfigurationException;
+import org.apache.tinkerpop.gremlin.driver.Client;
+import org.junit.Test;
+import org.springframework.util.Assert;
+
+public class GremlinFactoryUnitTest {
+
+    @Test
+    public void testGremlinFactory () {
+        GremlinFactory factory;
+        Client client;
+
+        try {
+            factory = new GremlinFactory(TestConstants.FAKE_ENDPOINT, null,
+                    TestConstants.FAKE_USERNAME, TestConstants.FAKE_PASSWORD);
+            client = factory.getGremlinClient();
+        } catch (Exception e) {
+            Assert.isTrue(e instanceof IllegalGremlinConfigurationException, "should export illegal exception");
```

Comment:
Can we separate this test case into 2 cases? seems have 2 different test purposes.  

About the exception check, there is ExpectedException or assertThrows etc. for JUnit, reference: https://junit.org/junit5/docs/current/user-guide/#writing-tests-assertions

Reply:
Thanks, will take a look and refine that.